test/e2e/storage/drivers/csi.go

```diff
@@ -110,54 +107,56 @@ func (h *hostpathCSIDriver) GetClaimSize() string {
 	return "5Gi"
 }
 
-func (h *hostpathCSIDriver) CreateDriver() {
+func (h *hostpathCSIDriver) CreateDriver(f *framework.Framework) (*testsuites.PerTestConfig, func()) {
 	By(fmt.Sprintf("deploying %s driver", h.driverInfo.Name))
-	f := h.driverInfo.Config.Framework
+	cancel := testsuites.StartPodLogs(f)
 	cs := f.ClientSet
 
 	// The hostpath CSI driver only works when everything runs on the same node.
 	nodes := framework.GetReadySchedulableNodesOrDie(cs)
 	nodeName := nodes.Items[rand.Intn(len(nodes.Items))].Name
-	h.driverInfo.Config.ClientNodeName = nodeName
+	config := &testsuites.PerTestConfig{
```

Comment:
its odd to me that `CreateDriver` would output a `PerTestConfig`. It may be cleaner to have whatever is calling `CreateDriver` to also call a `CreatePerTestConfig` as the next step and giving it the necessary inputs. 

`CreateDriver` can then just return whatever artifacts that only relate to the driver.

Reply:
`CreateDriver` always has been the "cluster is ready, set up test runtime resources" method. Going forward, drivers will be pre-installed during test cluster setup, but we might still need such a per-test setup function. Let's just rename `CreateDriver` to something that is a bit more generic. How about `PrepareTest`? What it does internally (install driver, deploy NFS server) is irrelevant at the API level, what matters is the result: cluster is ready, a per-test config describes it.

Reply:
got it. seems like the naming change will resolve this for me then

Reply:
Renamed.